api/v1alpha/api.proto

```diff
@@ -485,3 +503,63 @@ service PublicAPI {
         // the stream.
         rpc GetLogs(GetLogsRequest) returns (stream GetLogsResponse) {}
 }
+
+// FetchImagesRequest defines the request for FetchImages.
+message FetchImagesRequest {
+        // Type of the image, e.g. aci, docker, oci, etc.
+        Image type = 1;
+
+        // Resolvable names of the image, used for discovery, can be string, hash, URL, file path.
+        repeated string names = 2;
```

Comment:
I'll go with using the distribution concept from #2953. So image type is not needed for fetching. I left a comment for you there some days ago. Can you take a look at it?